### DISCUSSION TITLE
Label property

### DISCUSSION THREAD

**Discussion by yas-f** [chhoumann/quickadd#661] [Q&A]
Hey - thanks for making my favourite obsidian plugin.

Just wondering if I can modify the json file at all to add a label property to the menu. 

I have used the macros a lot and use menus to trigger or skip options (other macros). But I want to add a question as the menu label eg Would you like to proceed? as the label and for the menu options to be yes (triggering the next macro) or to skip.

How can I make the menu title ('name' property) show in the notice so I can save it as a question? 

Thanks!

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that, and thanks for the kind words.

There's no label/title property on the Multi (folder) menu, so editing data.json won't get you there. But for a yes/skip gate there are cleaner routes:

The direct one: replace the menu with a small user script step in your macro that asks the question and triggers the next macro on yes:

```js
module.exports = async ({ quickAddApi }) => {
  const proceed = await quickAddApi.yesNoPrompt("Would you like to proceed?");
  if (proceed) await quickAddApi.executeChoice("Name of next macro");
};
```

If you'd rather keep a list-style menu, format syntax can put the question on a suggester inside a choice: `{{VALUE:Yes,Skip|label:Would you like to proceed?}}` shows the two options with your question as the label. And for fully custom menus in a script, there's `quickAddApi.suggester(["Yes", "Skip"], [true, false])`.

Both `yesNoPrompt` and `executeChoice` are documented here: https://quickadd.obsidian.guide/docs/QuickAddAPI
